utility/src/main/java/org/oppia/android/util/parser/BlurTransformation.kt

```diff
@@ -0,0 +1,56 @@
+package org.oppia.android.util.parser
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.renderscript.Allocation
+import android.renderscript.Element
+import android.renderscript.RenderScript
+import android.renderscript.ScriptIntrinsicBlur
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
+import java.security.MessageDigest
+
+class BlurTransformation(context: Context) : BitmapTransformation() {
```

Comment:
@anandwana001 provided me this file. I didn't need to update Gradle, and I don't know about Bazel.

Reply:
@anandwana001 can you provide some context on where this file originated & the other questions above?

Reply:
Using this tutorial for reference - https://futurestud.io/tutorials/glide-custom-transformation

Reply:
We need to rewrite this based on the API--we shouldn't be exactly copying code from a site like that. We can use tutorials as a reference, but we should be actually typing out the code & understanding it, which means using our own variable names, structures, and documentation. @viktoriias can you please take a pass on rewriting this class based on existing documentation & references? Please make sure to cite any resources that you use in the documentation comment.

Further, how do we know this will work with Bazel on KK? Per the renderscript flag will we also need to enable a compile-time argument somewhere in the build pipeline for KK with Bazel?

Reply:
I added some suggestions re: rewriting.

We do need to test this on a KitKat emulator, and with Bazel before we can check it in. @viktoriias can you try running this with a KitKat emulator to see if the blur works? For Bazel, you can download the APK from [checks](https://github.com/oppia/oppia-android/pull/2422/checks?check_run_id=1746665207) at the top of the PR (look at the 'Artifacts' drop down--there's an APK built via Bazel that you can install on an emulator or real device using ``adb install``). Please also check that this build works correctly with blurring for both KK & non-KK Android versions.

Reply:
KitKat schreenshot:
![Screenshot_1611422658](https://user-images.githubusercontent.com/10843277/105609455-4de38f00-5d5e-11eb-9567-2b8c29508fdf.png)

Reply:
> I added some suggestions re: rewriting.
> 
> We do need to test this on a KitKat emulator, and with Bazel before we can check it in. @viktoriias can you try running this with a KitKat emulator to see if the blur works? For Bazel, you can download the APK from [checks](https://github.com/oppia/oppia-android/pull/2422/checks?check_run_id=1746665207) at the top of the PR (look at the 'Artifacts' drop down--there's an APK built via Bazel that you can install on an emulator or real device using `adb install`). Please also check that this build works correctly with blurring for both KK & non-KK Android versions.

@BenHenning When I tried to drag and drop the `oppia.apk` file to different emulators, I'm getting one of these errors:
`INSTALL_FAILED_VERSION_DOWNGRADE`
or
`The APK failed to install. Error: Could not parse error string`

Reply:
I got the APK to work now, and there is no blur..
![Screenshot_1611721377](https://user-images.githubusercontent.com/10843277/105942954-1baf8700-6015-11eb-8fd5-4ce314d99f06.png)

Reply:
@viktoriias ah. Regarding the downgrade, you can fix that by forcing the adb app install using '-r' if you run into this in the future. You'll likely see compatibility issues when switching the APK between Gradle & Bazel versions.

Regarding the blurring, I suspect that we will need to somehow enable render script for Bazel builds. I'll need to dig into this a bit.

Reply:
Ah, apparently '-r' doesn't properly install in DOWNGRADE situations. In that case, you can just uninstall the app on the device before running adb install.

Reply:
Uninstalling the app on the device worked for me, we should include these instructions in oppia wiki page.
There was a small bug in `BlurTransformation`. Updated version works with Bazel now: 
![Screenshot_1611805542](https://user-images.githubusercontent.com/10843277/106087529-10259400-60d9-11eb-89fe-69492a52f529.png)

Reply:
Looks great! Thanks!